File: est-features/est-features-ai/est-features-ai-impl/src/main/java/ltd/idcu/est/features/ai/impl/Ai.java
package ltd.idcu.est.features.ai.impl;

import ltd.idcu.est.features.ai.api.AiAssistant;

public class Ai {
    
    private static final AiAssistant INSTANCE = new DefaultAiAssistant();
    
    private Ai() {
    }
    
    public static AiAssistant get() {
        return INSTANCE;
    }
    
    public static String quickRef(String topic) {
        return INSTANCE.getQuickReference(topic);
    }
    
    public static String bestPractice(String category) {
        return INSTANCE.getBestPractice(category);
    }
    
    public static String tutorial(String topic) {
        return INSTANCE.getTutorial(topic);
    }
    
    public static String prompt(String templateName, java.util.Map<String, String> variables) {
        return INSTANCE.generatePrompt(templateName, variables);
    }
    
    public static String suggest(String requirement) {
        return INSTANCE.suggestCode(requirement);
    }
    
    public static String explain(String code) {
        return INSTANCE.explainCode(code);
    }
    
    public static String optimize(String code) {
        return INSTANCE.optimizeCode(code);
    }
}

File: est-admin-ui/src/api/ai.ts
import request from '@/utils/request'

export interface PromptTemplate {
  name: string
  category: string
  description: string
  template: string
  requiredVariables: Record<string, string>
}

export function chat(message: string) {
  return request.post<string>('/admin/api/ai/chat', null, {
    params: { message }
  })
}

export function generateCode(requirement: string) {
  return request.post<string>('/admin/api/ai/code/generate', null, {
    params: { requirement }
  })
}

export function suggestCode(requirement: string) {
  return request.post<string>('/admin/api/ai/code/suggest', null, {
    params: { requirement }
  })
}

export function explainCode(code: string) {
  return request.post<string>('/admin/api/ai/code/explain', null, {
    params: { code }
  })
}

export function optimizeCode(code: string) {
  return request.post<string>('/admin/api/ai/code/optimize', null, {
    params: { code }
  })
}

export function getQuickReference(topic: string) {
  return request.get<string>('/admin/api/ai/reference', {
    params: { topic }
  })
}

export function getBestPractice(category: string) {
  return request.get<string>('/admin/api/ai/bestpractice', {
    params: { category }
  })
}

export function getTutorial(topic: string) {
  return request.get<string>('/admin/api/ai/tutorial', {
    params: { topic }
  })
}

export function listTemplates() {
  return request.get<PromptTemplate[]>('/admin/api/ai/templates')
}

export function generatePrompt(data: {
  templateName: string
  variableNames?: string
  variableValues?: string
}) {
  return request.post<string>('/admin/api/ai/templates/generate', null, {
    params: data
  })
}
